{{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/templates/forms/input.ts

```diff
@@ -1,78 +1,92 @@
+      // (this should never happen!)
+      if (!currentInput) {
+        return;
+      }
 
-    if (currentInput.classList.contains("autofilled")) {
-      currentInput.classList.remove("autofilled");
-    }
-  },
-  init() {
-    // We're using the animationstart event to detect when the browser has autofilled
-    const currentInput = this.$refs.input as HTMLInputElement;
-    currentInput.addEventListener("animationstart", (e) => {
-      switch (e.animationName) {
-        case "autofill-start":
+      // if the input is focused, or has a value, set the active flag to true
+      // otherwise, set it to false
+      switch (e.type) {
+        case "focus":
+        case "input":
           this.active = true;
-          currentInput.classList.add("autofilled");
           break;
-        case "autofill-cancel":
-          if (document.activeElement === currentInput) {
-            this.active = true;
-          } else if (this.value === "") {
-            this.active = false;
-          }
+        case "blur":
           if (!currentInput.value) {
-            currentInput.classList.remove("autofilled");
+            this.active = false;
+          } else {
+            this.active = true;
           }
           break;
         default:
       }
-    });
 
-    if (this.value === "None") {
-      this.value = "";
-    } else if (this.$refs !== undefined && "input" in this.$refs) {
-      // Toggle the focused state on an input when an initial value is set.
-      this.active = !this.active;
-    }
-    if (this.eventName !== "input") {
-      this.$watch("value", () => {
-        this.$dispatch(
-          this.eventName,
-          { value: this.value },
-        );
+      if (currentInput.classList.contains("autofilled")) {
+        currentInput.classList.remove("autofilled");
+      }
+    },
+    init() {
+      // We're using the animationstart event to detect when the browser has autofilled
+      const currentInput = this.$refs.input as HTMLInputElement;
+      currentInput.addEventListener("animationstart", (e) => {
+        switch (e.animationName) {
+          case "autofill-start":
+            this.active = true;
+            currentInput.classList.add("autofilled");
+            break;
+          case "autofill-cancel":
+            if (document.activeElement === currentInput) {
+              this.active = true;
+            } else if (this.value === "") {
+              this.active = false;
+            }
+            if (!currentInput.value) {
+              currentInput.classList.remove("autofilled");
+            }
+            break;
+          default:
+        }
       });
-    }
-  },
-});
 
-AlpineInstance.data("input", input as AlpineDataCallback);
+      if (this.value === "None") {
+        this.value = "";
+      } else if (this.$refs !== undefined && "input" in this.$refs) {
+        // Toggle the focused state on an input when an initial value is set.
+        this.active = !this.active;
+      }
```

Comment:
Could be a little cleaner IMO
this.value = this.value === "None" ? "" : this.value;

if (this.$refs?.input) {
  // Toggle the focused state on an input when an initial value is set.
  this.active = !this.active;
}